ggshield/core/git_shell.py

```diff
@@ -131,16 +131,23 @@ def is_valid_git_commit_ref(ref: str) -> bool:
     return True
 
 
-def get_list_commit_SHA(commit_range: str) -> List[str]:
+def get_list_commit_SHA(
+    commit_range: str, max_count: Optional[int] = None
+) -> List[str]:
     """
     Retrieve the list of commit SHA from a range.
     :param commit_range: A range of commits (ORIGIN...HEAD)
+    :param max_count: If set, limits the number of SHA returned to this amount
```

Comment:
We may add a small note to tell that the count starts from the ORIGIN. This may not be obvious to the reader.

Reply:
Actually it does not: it's the max-count most recent commits :)